[0.5.0-rc2] Nomad agent invalid memory address or nil pointer dereference

Hi

Trying out an upgrade to Nomad v0.5.0-rc2 from v0.4.1 and got this:
```
==> Starting Nomad agent...
==> Nomad agent configuration:

                 Atlas: <disabled>
                Client: true
             Log Level: INFO
                Region: global (DC: dc1)
                Server: false
               Version: 0.5.0rc2

==> Nomad agent started! Log data will stream in below:

    2016/11/15 08:23:22.591557 [INFO] client: using state directory /data/1/nomad/client
    2016/11/15 08:23:22.591595 [INFO] client: using alloc directory /data/1/nomad/alloc
    2016/11/15 08:23:22.591861 [INFO] fingerprint.cgroups: cgroups are available
    2016/11/15 08:23:22.594080 [INFO] fingerprint.consul: consul agent is available
    2016/11/15 08:23:24.614525 [INFO] client: Node ID "a11c130f-0511-7038-42b0-a8e15f4b1604"
    2016/11/15 08:23:24.618594 [INFO] client.consul: discovered following Servers: xx.xx.xx.149:4647,xx.xx.xx.148:4647,xx.xx.xx.82:4647,xx.xx.xx.81:4647,xx.xx.xx.80:4647
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5587da]

goroutine 56 [running]:
panic(0xfbb920, 0xc4200100b0)
	/opt/go/src/runtime/panic.go:500 +0x1a1
github.com/hashicorp/nomad/client.(*Client).runAllocs(0xc42023bb80, 0xc4202be8a0)
	/opt/gopath/src/github.com/hashicorp/nomad/client/client.go:1396 +0x6aa
github.com/hashicorp/nomad/client.(*Client).run(0xc42023bb80)
	/opt/gopath/src/github.com/hashicorp/nomad/client/client.go:927 +0x119
created by github.com/hashicorp/nomad/client.NewClient
	/opt/gopath/src/github.com/hashicorp/nomad/client/client.go:266 +0xc4a
    Loaded configuration from /tmp/nomad.cfg
```
/tmp/nomad.cfg:
```
datacenter = "dc1"
client {
  enabled = true
  node_class = "slave"
}
data_dir = "/data/1/nomad"
bind_addr = "0.0.0.0"
telemetry {
  "statsd_address" = "127.0.0.1:8125"
}
advertise {
  http = "xx.xx.xx.148:4646"
  rpc = "xx.xx.xx.148:4647"
  serf = "xx.xx.xx.148:4648"
}
```
Commandline:
`nomad agent -client -config=/tmp/nomad.cfg`

--- a/client/client.go
+++ b/client/client.go
@@ -1393,7 +1393,7 @@ func (c *Client) runAllocs(update *allocUpdates) {
 		// previous allocation
 		var prevAllocDir *allocdir.AllocDir
 		tg := add.Job.LookupTaskGroup(add.TaskGroup)
-		if tg != nil && tg.EphemeralDisk.Sticky == true && ar != nil {
+		if tg != nil && tg.EphemeralDisk != nil && tg.EphemeralDisk.Sticky == true && ar != nil {
 			prevAllocDir = ar.GetAllocDir()
 		}
 
@@ -1432,7 +1432,7 @@ func (c *Client) blockForRemoteAlloc(alloc *structs.Allocation) {
 	}
 
 	// Wait for the remote previous alloc to be terminal if the alloc is sticky
-	if tg.EphemeralDisk.Sticky {
+	if tg.EphemeralDisk != nil && tg.EphemeralDisk.Sticky {
 		c.logger.Printf("[DEBUG] client: blocking alloc %q for previous allocation %q", alloc.ID, alloc.PreviousAllocation)
 		// Block until the previous allocation migrates to terminal state
 		prevAlloc, err := c.waitForAllocTerminal(alloc.PreviousAllocation)
@@ -1509,7 +1509,7 @@ func (c *Client) migrateRemoteAllocDir(alloc *structs.Allocation, allocID string
 
 	// Skip migration of data if the ephemeral disk is not sticky or
 	// migration is turned off.
-	if !tg.EphemeralDisk.Sticky || !tg.EphemeralDisk.Migrate {
+	if tg.EphemeralDisk == nil || !tg.EphemeralDisk.Sticky || !tg.EphemeralDisk.Migrate {
 		return nil, nil
 	}
 

